[CI] LogsdbTestSuiteIT class failing

**Build Scans:**
- [elasticsearch-intake #11369 / part2](https://gradle-enterprise.elastic.co/s/3mou4r3m52nzg)
- [elasticsearch-pull-request #36596 / part-2](https://gradle-enterprise.elastic.co/s/r5zg25d2f42wo)
- [elasticsearch-pull-request #36520 / part-2](https://gradle-enterprise.elastic.co/s/es24jxrznfpj6)
- [elasticsearch-pull-request #36296 / part-2](https://gradle-enterprise.elastic.co/s/empsus2dbb7jy)
- [elasticsearch-pull-request #36256 / part-2](https://gradle-enterprise.elastic.co/s/glqlkcropwkaw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:logsdb:yamlRestTest" --tests "org.elasticsearch.xpack.logsdb.LogsdbTestSuiteIT.test {yaml=/30_logsdb_default_mapping/create logsdb data stream with custom sorting and host object}" -Dtests.seed=8B80B4CBFE7E91A9 -Dtests.locale=sr-ME -Dtests.timezone=Europe/Tirane -Druntime.java=22
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.logsdb.LogsdbTestSuiteIT'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.logsdb.LogsdbTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [/30_logsdb_default_mapping:342]: got unexpected warning header [
	299 Elasticsearch-8.17.0-e53f30177f2ad8152f1b503782656cad9b6c12d2 "index template [logs-template] has index patterns [logs-nginx-prod] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
]

```

**Issue Reasons:**
- [8.x] 5 failures in class org.elasticsearch.xpack.logsdb.LogsdbTestSuiteIT (1.3% fail rate in 376 executions)
- [8.x] 4 failures in step part-2 (4.5% fail rate in 89 executions)
- [8.x] 4 failures in pipeline elasticsearch-pull-request (4.8% fail rate in 83 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/logsdb/src/yamlRestTest/resources/rest-api-spec/test/30_logsdb_default_mapping.yml
+++ b/x-pack/plugin/logsdb/src/yamlRestTest/resources/rest-api-spec/test/30_logsdb_default_mapping.yml
@@ -291,7 +291,7 @@ create logsdb data stream with custom sorting without host.name:
 
   - do:
       allowed_warnings:
-        - "index template [logs-template] has index patterns [logs-*-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
+        - "index template [logs-template] has index patterns [logs-http-prod] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logs-template
         body:
@@ -341,7 +341,7 @@ create logsdb data stream with custom sorting and host object:
 
   - do:
       allowed_warnings:
-        - "index template [logs-template] has index patterns [logs-*-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
+        - "index template [logs-template] has index patterns [logs-nginx-prod] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logs-template
         body:
@@ -400,7 +400,7 @@ create logsdb data stream with custom sorting and dynamically mapped host.name:
 
   - do:
       allowed_warnings:
-        - "index template [logs-template] has index patterns [logs-*-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
+        - "index template [logs-template] has index patterns [logs-kafka-qa] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logs-template
         body:
@@ -465,7 +465,7 @@ create logsdb data stream with custom sorting and host.name object:
 
   - do:
       allowed_warnings:
-        - "index template [logs-template] has index patterns [logs-*-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
+        - "index template [logs-template] has index patterns [logs-nginx-qa] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logs-template
         body:
@@ -521,7 +521,7 @@ create logsdb data stream with default sorting on malformed host.name:
 
   - do:
       allowed_warnings:
-        - "index template [logs-template] has index patterns [logs-*-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
+        - "index template [logs-template] has index patterns [logs-win-prod] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logs-template
         body:
@@ -585,7 +585,7 @@ create logsdb data stream with custom sorting and host.name date field:
 
   - do:
       allowed_warnings:
-        - "index template [logs-template] has index patterns [logs-*-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
+        - "index template [logs-template] has index patterns [logs-http-prod] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logs-template
         body:
@@ -638,7 +638,7 @@ create logsdb data stream with custom sorting and missing host.name field mappin
 
   - do:
       allowed_warnings:
-        - "index template [logs-template] has index patterns [logs-*-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
+        - "index template [logs-template] has index patterns [logs-http-qa] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logs-template
         body:
@@ -690,7 +690,7 @@ create logsdb data stream with custom sorting and host.name field without doc va
 
   - do:
       allowed_warnings:
-        - "index template [logs-template] has index patterns [logs-*-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
+        - "index template [logs-template] has index patterns [logs-http-dev] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logs-template
         body:
@@ -732,7 +732,7 @@ create logsdb data stream with incompatible ignore_above on host.name:
 
   - do:
       allowed_warnings:
-        - "index template [logsdb-index-template-ignore-above] has index patterns [logsdb-ignore-above] matching patterns from existing older templates [global]"
+        - "index template [logsdb-index-template-ignore-above] has index patterns [logsdb-ignore-above] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logsdb-index-template-ignore-above] will take precedence during new index creation"
       indices.put_index_template:
         name: logsdb-index-template-ignore-above
         body:
@@ -780,7 +780,7 @@ create logsdb data stream with no sorting and host.name as text:
 
   - do:
       allowed_warnings:
-        - "index template [logsdb-index-template-non-keyword] has index patterns [logsdb-non-keyword] matching patterns from existing older templates [global]"
+        - "index template [logsdb-index-template-non-keyword] has index patterns [logsdb-non-keyword] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logsdb-index-template-non-keyword] will take precedence during new index creation"
       indices.put_index_template:
         name: logsdb-index-template-non-keyword
         body:
@@ -814,7 +814,7 @@ create logsdb data stream without index sorting and ignore_above on host.name:
 
   - do:
       allowed_warnings:
-        - "index template [logsdb-index-template-ignore-above-override] has index patterns [logsdb-ignore-above-override] matching patterns from existing older templates [global]"
+        - "index template [logsdb-index-template-ignore-above-override] has index patterns [logsdb-ignore-above-override] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logsdb-index-template-ignore-above-override] will take precedence during new index creation"
       indices.put_index_template:
         name: logsdb-index-template-ignore-above-override
         body:
@@ -860,7 +860,7 @@ create logsdb data stream with host.name as alias and sorting on it:
 
   - do:
       allowed_warnings:
-        - "index template [logsdb-index-template-alias] has index patterns [logsdb-alias] matching patterns from existing older templates [global]"
+        - "index template [logsdb-index-template-alias] has index patterns [logsdb-alias] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logsdb-index-template-alias] will take precedence during new index creation"
       indices.put_index_template:
         name: logsdb-index-template-alias
         body:
@@ -898,7 +898,7 @@ create logsdb data stream with multi-fields on host.name:
 
   - do:
       allowed_warnings:
-        - "index template [logsdb-index-template-multi-fields] has index patterns [logsdb-multi-fields] matching patterns from existing older templates [global]"
+        - "index template [logsdb-index-template-multi-fields] has index patterns [logsdb-multi-fields] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logsdb-index-template-multi-fields] will take precedence during new index creation"
       indices.put_index_template:
         name: logsdb-index-template-multi-fields
         body:
@@ -945,7 +945,7 @@ create logsdb data stream with multi-fields on host.name and no sorting:
 
   - do:
       allowed_warnings:
-        - "index template [ logsdb-no-sort-multi-fields-template ] has index patterns [logsdb-no-sort-multi-fields] matching patterns from existing older templates [global]"
+        - "index template [logsdb-no-sort-multi-fields-template] has index patterns [logsdb-no-sort-multi-fields] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logsdb-no-sort-multi-fields-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logsdb-no-sort-multi-fields-template
         body:
@@ -980,7 +980,7 @@ create logsdb data stream with custom empty sorting:
 
   - do:
       allowed_warnings:
-        - "index template [logs-template] has index patterns [logs-*-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
+        - "index template [logs-template] has index patterns [logs-http-empty] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logs-template
         body:
@@ -1027,7 +1027,7 @@ create logsdb data stream with custom sorting on timestamp:
 
   - do:
       allowed_warnings:
-        - "index template [logs-template] has index patterns [logs-*-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
+        - "index template [logs-template] has index patterns [logs-http-dev] matching patterns from existing older templates [global] with patterns (global => [*]); this template [logs-template] will take precedence during new index creation"
       indices.put_index_template:
         name: logs-template
         body:

